subworkflows/nf-core/bam_downsample_samtools/main.nf

```diff
@@ -0,0 +1,54 @@
+include { SAMTOOLS_DEPTH } from '../../../modules/nf-core/samtools/depth'
+include { GAWK           } from '../../../modules/nf-core/gawk'
+include { SAMTOOLS_VIEW  } from '../../../modules/nf-core/samtools/view'
+
+workflow BAM_DOWNSAMPLE_SAMTOOLS {
+
+    take:
+    ch_bam    // channel: [ [id], bam, bai ]
+    ch_fasta  // channel: [ [genome], fasta ]
+    depth     // channel: [ depth ]
+
+    main:
+    ch_versions      = Channel.empty()
+
+    // Compute mean depth
+    SAMTOOLS_DEPTH(ch_bam, [[], []])
+    ch_versions = ch_versions.mix(SAMTOOLS_DEPTH.out.versions.first())
+
+    // Use GAWK to get mean depth
+    GAWK(SAMTOOLS_DEPTH.out.tsv, [])
+    ch_versions = ch_versions.mix(GAWK.out.versions.first())
+
+    // Compute downsampling factor
+    ch_mean_depth = GAWK.out.output
+        .splitCsv(header: false, sep:'\t')
+        .map{ metaI, row ->
+            [ metaI, row[0] as Float ]
```

Comment:
What does `metaI` stand for? If you need to name this something other than meta, maybe the naming could be a bit more descriptive.

Reply:
It was for the `id` field. But it isn't any more needed.